master/internal/api_workspace.go

```diff
@@ -185,13 +237,8 @@ func (a *apiServer) PostWorkspace(
 func (a *apiServer) PatchWorkspace(
 	ctx context.Context, req *apiv1.PatchWorkspaceRequest,
 ) (*apiv1.PatchWorkspaceResponse, error) {
-	user, err := a.CurrentUser(ctx, &apiv1.CurrentUserRequest{})
-	if err != nil {
-		return nil, err
-	}
-
-	// Verify current workspace exists and can be edited.
-	currWorkspace, err := a.GetWorkspaceByID(req.Id, user.User.Id, true)
+	currWorkspace, currUser, err := a.getWorkspaceAndCheckCanDoActions(ctx, req.Id, true,
+		workspace.AuthZProvider.Get().CanSetWorkspacesName)
```

Comment:
in case we add more fields in the future, I'd prefer this to be more like the `PatchProject` function and check access inside of the `if req.Workspace.Name != nil ... { ... }` block